File: brawlbuddy-frontend/src/types/Brawler.ts
// Brawler types based on backend API models
export interface StarPower {
  id: number;
  name: string;
  imageUrl?: string; // Example from Brawlify, adjust if API provides
  description?: string;
}

export interface Gadget {
  id: number;
  name: string;
  imageUrl?: string; // Example from Brawlify, adjust if API provides
  description?: string;
}

export interface BrawlerRarity {
  id?: number; // ID might not always be present or needed from all APIs
  name: string;
  color: string;
}

export interface BrawlerClass {
  id?: number; // ID might not always be present or needed
  name: string;
}

export interface Brawler {
  id: number;
  name: string;
  description?: string;
  imageUrl?: string; // Main image for the brawler
  imageUrl2?: string; // Secondary image or icon
  imageUrl3?: string; // Pin or other asset
  rarity: BrawlerRarity;
  class: BrawlerClass;
  starPowers: StarPower[];
  gadgets: Gadget[];
  // Gears can be added if the API supports it and it's in the C# model
  // gears?: Gear[]; 
}

// This is what the backend's /api/brawlers endpoint returns
export interface AllBrawlersResponse {
  brawlers: Brawler[];
  count: number;
}

File: BrawlBuddy.Api/Models/Brawler.cs
// Brawler model based on Brawl Stars API structure

namespace BrawlBuddy.Api.Models;

public class Brawler
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? ImageUrl { get; set; }
    public BrawlerRarity? Rarity { get; set; }
    public BrawlerClass? Class { get; set; }
    public List<BrawlerStarPower> StarPowers { get; set; } = new();
    public List<BrawlerGadget> Gadgets { get; set; } = new();
}

public class BrawlerRarity
{
    public int? Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Color { get; set; } = string.Empty;
}

public class BrawlerClass
{
    public int? Id { get; set; }
    public string Name { get; set; } = string.Empty;
}

public class BrawlerStarPower
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? ImageUrl { get; set; }
}

public class BrawlerGadget
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? ImageUrl { get; set; }
}